src/board.js

```diff
@@ -0,0 +1,73 @@
+import { useState } from 'react'
+import { Square } from './square.js'
+
+const board = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+export function Board(props) {
+  const [squareColors, setSquareColors] = useState(Array(board.length * board.length).fill('grey'))
+  const [clickedSquare, setClickedSquare] = useState(null)
+
+  const handleSquareClick = (i) => {
+    const squares_slice = squareColors.slice()
```

Comment:
I am pretty sure doing `squareColors.slice()` with no argument doesn't actually do anything.

Reply:
It copies it. The tutorial recommended doing this instead of updating the array directly

Reply:
Hmm... 🤷 seems weird to me still. Especially cause you just call `set...` right after

Reply:
[See here](https://reactjs.org/tutorial/tutorial.html#:~:text=Note%20how%20in%20handleClick%2C%20we%20call%20.slice()%20to%20create%20a%20copy%20of%20the%20squares%20array%20to%20modify%20instead%20of%20modifying%20the%20existing%20array.%20We%20will%20explain%20why%20we%20create%20a%20copy%20of%20the%20squares%20array%20in%20the%20next%20section.) @hogiyogi597